grails-app/services/org/pih/warehouse/requisition/RequisitionService.groovy

```diff
@@ -160,6 +160,43 @@ class RequisitionService {
         }
     }
 
+    /**
+     * Get all stock lists for given multiple origins and destinations
+     *
+     * @param origins
+     * @param destinations
+     * @return
+     */
+    List<Requisition> getRequisitionTemplates(List<Location> origins, List<Location> destinations) {
+        return Requisition.createCriteria().list {
+            eq("isTemplate", Boolean.TRUE)
+            or {
+                if (origins) {
+                    'in'("origin", origins)
+                }
+                if (destinations) {
+                    'in'("destination", destinations)
+                }
+            }
+        }
+    }
+
+    /**
+     * Get all items for given requisitions
+     * @param List<Location> origins
+     * @param List<Location> destinations
+     * @return
+     */
+    List<Requisition> getRequisitionTemplatesItems(List<Requisition> requisitions) {
+        def requisitionItems = RequisitionItem.createCriteria().list() {
+         if (requisitions) {
+                'in'("requisition", requisitions)
+            }
+        }
+
+        return requisitionItems.sort { a,b -> a.product.category <=> b.product.category ?: a.product.name <=> b.product.name }
```

Comment:
Sorting should be handled in the query. I believe you can do something like this to implement the same sorting policy.
```
def requisitionItems = RequisitionItem.createCriteria().list() {
    requisition { 
        eq("isTemplate", Boolean.TRUE)
        eq("isPublished", Boolean.TRUE)   
        if (requisitions) {
            'in'("destination", requisitions)
        }
    }
    product {
        category {             
            order("name", "asc")
        }
        order("name", "asc")
    }
}
```

Reply:
Also, can you ask Jesse/Kelsey if the export should return published and/or unpublished stocklist items.